studio/actions/cancel-event.tsx

```diff
@@ -0,0 +1,91 @@
+import { ErrorOutlineIcon, WarningOutlineIcon } from "@sanity/icons";
+import { useToast, Card, Text } from "@sanity/ui";
+import { useState } from "react";
+import { DocumentActionProps, DocumentActionComponent, useDocumentOperation } from "sanity";
+import { Event } from "../models/sanity.model";
+import { sendEmailEventCanceled } from "../lib/event-email";
+
+export const CancelAction: DocumentActionComponent = (props: DocumentActionProps) => {
+  const toast = useToast();
+
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const { patch, unpublish } = useDocumentOperation(props.id, props.type);
+
+  const publishedEvent = props.published as Event | null;
+  return {
+    disabled: !publishedEvent,
+    label: "Avlys arrangement",
+    tone: "critical",
+    icon: ErrorOutlineIcon,
+    onHandle: () => {
+      setDialogOpen(true);
+    },
+    dialog: dialogOpen && {
+      type: "confirm",
+      onCancel: props.onComplete,
+      onConfirm: async () => {
+        if (publishedEvent) {
+          const result = await handleEventCancel(publishedEvent);
+          if (result) {
+            if (!publishedEvent.title.endsWith("Avlyst")) {
+              patch.execute([{ set: { title: `${publishedEvent.title} | Avlyst` } }]);
+            }
+
+            unpublish.execute();
+
+            toast.push({
+              status: "success",
+              title: "Arrangementet er avlyst",
+            });
+          } else {
+            toast.push({
+              status: "error",
+              title: "En feil oppstod ved avlysing av arrangementet",
+            });
+          }
+        }
+        props.onComplete();
+      },
+      message: (
+        <Card padding={4}>
+          <Text>
+            <WarningOutlineIcon
+              style={{
+                color: "red",
+                fontSize: "24px",
+                marginRight: "3px",
+                marginLeft: "1px",
+              }}
+            />
+            <span>
+              Vennligst bekreft at du ønsker å avlyse arrangementet. En e-post avlysing blir sendt
+              til alle påmeldte.
```

Comment:
"e-post avlysning" i to ord blir litt feil, kanskje heller skrive "Alle påmeldte blir varslet på e-post om at arrangementet er avlyst." eller noe?

Reply:
Yes enig! 🙌